features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/sql/mssql.sql

```diff
@@ -2279,16 +2279,45 @@ CREATE TABLE AM_REVISION (
   UNIQUE(REVISION_UUID)
 )ENGINE INNODB;
 
-IF NOT EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[AM_SCOPE_BINDING]') AND TYPE IN (N'U'))
+IF NOT EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[AM_DEPLOYMENT_REVISION_MAPPING]') AND TYPE IN (N'U'))
 CREATE TABLE AM_DEPLOYMENT_REVISION_MAPPING (
   NAME VARCHAR(255) NOT NULL,
+  VHOST VARCHAR(255) NOT NULL,
   REVISION_UUID VARCHAR(255) NOT NULL,
   DISPLAY_ON_DEVPORTAL BIT DEFAULT 0,
   DEPLOYED_TIME DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
   PRIMARY KEY (NAME, REVISION_UUID),
   FOREIGN KEY (REVISION_UUID) REFERENCES AM_REVISION(REVISION_UUID) ON UPDATE CASCADE ON DELETE CASCADE
 )ENGINE INNODB;
 
+-- Gateway Environments Table --
+IF NOT EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[AM_GATEWAY_ENVIRONMENT]') AND TYPE IN (N'U'))
+CREATE TABLE AM_GATEWAY_ENVIRONMENT (
+  ID INTEGER IDENTITY,
+  UUID VARCHAR(45) NOT NULL,
+  NAME VARCHAR(255) NOT NULL,
+  TENANT_DOMAIN VARCHAR(255),
+  DISPLAY_NAME VARCHAR(255) NULL,
+  DESCRIPTION VARCHAR(1023) NULL,
+  UNIQUE (NAME, TENANT_DOMAIN),
+  UNIQUE (UUID),
+  PRIMARY KEY (ID)
+)ENGINE INNODB;
+
+-- Virtual Hosts Table --
+IF NOT EXISTS (SELECT * FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[DBO].[AM_GW_VHOST]') AND TYPE IN (N'U'))
+CREATE TABLE AM_GW_VHOST (
+  GATEWAY_ENV_ID INTEGER,
+  HOST VARCHAR(255) NOT NULL,
+  HTTP_CONTEXT VARCHAR(255) NULL,
+  HTTP_PORT VARCHAR(5) NOT NULL,
+  HTTPS_PORT VARCHAR(5) NOT NULL,
+  WS_PORT VARCHAR(5) NOT NULL,
+  WSS_PORT VARCHAR(5) NOT NULL,
+  FOREIGN KEY (GATEWAY_ENV_ID) REFERENCES AM_GATEWAY_ENVIRONMENT(ID) ON UPDATE CASCADE ON DELETE CASCADE,
+  UNIQUE (GATEWAY_ENV_ID, HOST)
+)ENGINE INNODB;
+
```

Comment:
Shall we add a PK to this table ? When creating mssql replications the tables must have a PK

Reply:
fixed with